github/event_types.go

```diff
@@ -9,6 +9,10 @@ package github
 
 import "encoding/json"
 
+type GitHubEvent interface {
```

Comment:
Please add a comment to this interface describing why we are choosing to do this.
Feel free to link to the issue where a long discussion took place.  :smile: